.changelog/unreleased/improvements/1556-arch-doc.md

```diff
@@ -0,0 +1,6 @@
+- Add [architecture.md][arch-doc] doc that gives a high-level overview of the structure of the codebase.
+- Add some module-level documentation 
+ ([#1556][1556])
+
+[1556]: https://github.com/informalsystems/ibc-rs/pulls/1556
+[arch-doc]: https://github.com/informalsystems/ibc-rs/blob/2f9bdcd1de847b8dc5f21bc7b1df1dc177589eaa/docs/architecture/architecture.md
```

Comment:
It's possible this link will go 404 one you merge & delete your branch?
If yes, then we can consider simply not putting any link at all.